lib/java/AdapterLibrary/src/main/kotlin/com/vmware/aria/operations/AdapterLogger.kt

```diff
@@ -0,0 +1,257 @@
+/*
+ * Copyright 2023 VMware, Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+@file:JvmName("AdapterLogger")
+
+package com.vmware.aria.operations
+
+import org.apache.logging.log4j.Level
+import org.apache.logging.log4j.LogManager
+import org.apache.logging.log4j.Logger
+import org.apache.logging.log4j.core.LoggerContext
+import org.apache.logging.log4j.core.appender.RollingFileAppender
+import org.apache.logging.log4j.core.appender.rolling.CompositeTriggeringPolicy
+import org.apache.logging.log4j.core.appender.rolling.DefaultRolloverStrategy
+import org.apache.logging.log4j.core.appender.rolling.OnStartupTriggeringPolicy
+import org.apache.logging.log4j.core.appender.rolling.SizeBasedTriggeringPolicy
+import org.apache.logging.log4j.core.config.Configurator
+import org.apache.logging.log4j.core.layout.PatternLayout
+import org.apache.logging.log4j.util.StackLocatorUtil
+import java.io.File
+import kotlin.io.path.Path
+
+
+/**
+ * Sets up logging using the given parameters
+ *
+ * @param filename The name of the file to log to.
+ * @param backupFileCount The total number of backup files to retain. Defaults to 5.
+ * @param maxSize The maximum size in bytes of each file before the file
+ *  automatically rotates to a new one. Defaults to '0', which will
+ *  do no automatic rotation. May require calling the 'rotate()' function
+ *  manually to ensure logs do not become too large.
+ */
+@JvmOverloads
+fun setupLogging(
+    filename: String,
+    backupFileCount: Int = 5,
+    maxSize: Long = 0L,
+) {
+    // '/var/log' is the location of logs in the Adapter Container, and shouldn't
+    // be changed.
+    setupLogging(filename, backupFileCount, maxSize, "/var/log")
+}
+
+/**
+ * Internal constructor for testing purposes that sets up logging using the given parameters
+ *
+ * @param filename The name of the file to log to.
+ * @param backupFileCount The total number of backup files to retain. Defaults to 5.
+ * @param maxSize The maximum size in bytes of each file before the file
+ *  automatically rotates to a new one. Defaults to '0', which will
+ *  do no automatic rotation. Requires calling the 'rotate()' function
+ *  manually to ensure logs do not become too large.
+ *  @param filePath The location where the config file and logs reside in.
+ */
+internal fun setupLogging(
+    filename: String,
+    backupFileCount: Int,
+    maxSize: Long,
+    filePath: String,
+) {
+
+    val pattern = "%d [%t] %-5level: %msg%n%throwable"
```

Comment:
Would be useful to add a comment here with an example log that follows this pattern.